lib/semantics/resolve-names.cc

```diff
@@ -4988,7 +4994,8 @@ void ResolveNamesVisitor::CheckImport(
 }
 
 bool ResolveNamesVisitor::Pre(const parser::ImplicitStmt &x) {
-  return CheckNotInBlock("IMPLICIT") && ImplicitRulesVisitor::Pre(x);
+  return CheckNotInBlock("IMPLICIT") &&  // C1107
```

Comment:
2008-C806 says the specification part of a BLOCK construct must not contain an IMPLICIT statement.  However, 2018-C1107 does not include this restriction.  Unless some other clause addresses this, I believe this check should be removed.  (Compare the IMPORT statement, which is disallowed in a BLOCK construct in 2008 by C1210, but allowed in 2018 by R1109.)

Reply:
R1109 is what disallows an implicit-part in a block-specification-part, so the check is correct but the comment should be changed.

@CarolineConcatto, there are two TODOs in `parse-tree.h` after the definition of `BlockSpecificationPart`. This check completes the first one and I think you have taken care of everything in the second one. If that's the case, would you mind removing those as part of this pull request?

Reply:
It looks like Tim is right.  A _specification-construct_ does not include an _implicit-part_.  So the restriction remains even though it does not appear in C1107.

Reply:
From the 2 TODO I only implemented the second one. So I'll remove that one.
// TODO: error if any COMMON, EQUIVALENCE, INTENT, NAMELIST, OPTIONAL,
// VALUE, ENTRY, SAVE /common/, or statement function definition statement
// appears in a block-specification part (C1107, C1570).
Is that ok?
And I understand the IMPLICIT is allright as it is. Correct?